userspace/libscap/scap_vtable.h

```diff
@@ -223,6 +223,50 @@ struct scap_vtable {
 	 * @return the buffer space used, in bytes
 	 */
 	uint64_t (*get_max_buf_used)(struct scap_engine_handle engine);
+
+	/**
+	 * @brief get the list of all threads in the system, with their cpu usage
+	 * @param engine wraps the pointer to the engine-specific handle
+	 * @param procinfo_p pointer to pointer to the resulting list
+	 * @return SCAP_SUCCESS or a failure code
+	 *
+	 * `procinfo_p` must not be NULL, but `*procinfo_p` may be; the returned
+	 * list will be (re)allocated on demand
+	 */
+	int32_t (*get_threadlist)(struct scap_engine_handle engine, struct ppm_proclist_info **procinfo_p);
```

Comment:
Just a side question: is there any actual check that when a vtable is defined, all of its fields are actually non-NULL?
Or, we could do that if they're NULL, scap automatically sets the no-op for you. 
This way no-op vtables get much shorter (just a bunch of NULL and a name).

Reply:
It's C, so there are no checks and we're in full yolo mode. Somewhat more seriously, we'd probably want to skip the check for .next() anyway. Also, all the vtables are const and we can't modify them at runtime (we can copy them and modify the copy if we want).

The check for a full vtable would have to be manually written too (can't introspect the struct automatically).

All in all, it's not the highest priority for me right now, but I'm certainly not going to oppose a follow up PR that handles this better :)

Reply:
I will give this a look once everything is in place; thanks!